### PR TITLE
refactor(home): extract AllowanceView and DefaultView

### PR DESCRIPTION
**Pull request by lisabaut**
### Describe the changes you have made in this PR

This is a bigger refactoring which divides the Home screen (initial screen in PopUp) into two views:
- AllowanceView
- DefaultView

The refactoring required these steps:
- align all places where the `payments` for the component `TransactionTable` are created 
- => this fixes this ticket too https://github.com/getAlby/lightning-browser-extension/issues/1356 (correct @escapedcat ?)
- extend `Transaction` type with `publisherLink` as this is the main difference between the TransactionTable data in views for Allowance, Default (no Allowance), PublisherSingleView
- extracting components in two new folders `screens/Home/DefaultView` and `screens/Home/AllowanceView`

**⭐ PLEASE follow the commit history if you have problems reading the diff ⭐** 

---

### Link this PR to an issue [optional]

Closes https://github.com/getAlby/lightning-browser-extension/issues/1359
Closes https://github.com/getAlby/lightning-browser-extension/issues/1356

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### How has this been tested?

- On stacker news with setting a budget, editing it, deleting it
- adding example metatag to a website and check `keysend` (`<meta name="lightning" content="method=keysend; address=030a58b8653d32b99200a2334cfe913e51dc7d155aa0116c176657a4f1722677a3; customKey=696969; customValue=3wQCCrfOAMYNzOh1sL05" />`)

Would be cool to have a list of sites/pages to test in general.
And I can add more tests to this PR too of course. 

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [ ] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Review by escapedcat — APPROVED** [review]

File: src/app/screens/Home/AllowanceView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          <IconButton
+            onClick={props.onGoBack}
+            icon={<CaretLeftIcon className="w-4 h-4" />}
+          />
+        }
+      />
+      <div className="mx-4">
+        <PublisherCard
+          title={props.allowance.name}
+          description={""}
```
[the remark this answers is not in the document]
Comment (lisabaut) [reply src/app/screens/Home/AllowanceView/index.tsx]: done

**Review by bumi — CHANGES_REQUESTED** [review]
---package-lock.json must not be in the repo. we use yarn.--- 
github navigation fu*up sorry.

File: src/app/screens/Home/index.tsx [hunk 2]
```diff
@@ -1,480 +1,100 @@
...
-        active: true,
-        currentWindow: true,
-      });
-      const [currentTab] = tabs;
-      const url = new URL(currentTab.url as string);
-      setCurrentUrl(url);
-      const result = await api.getAllowance(url.host);
+      setLoadingAllowance(true);
+
+      if (!currentUrl) throw new Error("No established browser connection");
```
Comment (bumi) [inline src/app/screens/Home/index.tsx:19]: what is this about? what means "No established browser connection" ? 

isn't also loadAllowance only called if there is currentUrl?

File: src/app/screens/Home/index.tsx [hunk 3]
```diff
@@ -1,480 +1,100 @@
...
-            />
-          }
-        />
-      )}
-      {renderPublisherCard()}
-      {allowance ? renderAllowanceView() : renderDefaultView()}
-    </div>
-  );
-}
+  return <DefaultView currentUrl={currentUrl} lnDataFromCurrentTab={lnData} />;
```
Comment (bumi) [inline src/app/screens/Home/index.tsx:97]: so we keep the lnData handling under the Default view?

**Review by bumi — COMMENTED** [review]
With this change we no longer show the tipping option if we have an allowance. 

The "send satoshi" option will only be available for sites without allowance. 
Once a user clicks on a `lightning:` link on a page there will be an allowances saved. 
So I think this restriction is too much? 

@reneaaron what have been thoughts and cases here?

File: src/app/screens/Home/index.tsx (hunk 2 above)
Comment (lisabaut) [reply src/app/screens/Home/index.tsx:20]: This is for typescript. I added a comment to make this more clear

File: src/app/screens/Home/index.tsx (hunk 3 above)
Comment (lisabaut) [reply src/app/screens/Home/index.tsx:98]: Currently yes.

Can you provide me with a test case where an allowance is set and lnurl is integrated too? 

Then I will add the lnurl PublisherCard to the AllowanceView too

Comment (bumi) [reply src/app/screens/Home/index.tsx:98]: see getalby.com 

visit it without allowance and with allowance (e.g. after logged in with lightning)

File: src/app/components/TransactionsTable/index.tsx [hunk 4]
```diff
@@ -52,7 +52,19 @@ export default function TransactionsTable({ transactions }: Props) {
...
                       >
-                        {tx.title || "\u00A0"}
+                        <p className="truncate">
+                          {tx.publisherLink ? (
+                            <a
+                              target="_blank"
+                              href={tx.publisherLink}
+                              rel="noreferrer"
+                            >
+                              {tx.title || "\u00A0"}
```
[the remark this answers is not in the document]
Comment (lisabaut) [reply src/app/components/TransactionsTable/index.tsx]: Thanks, done

File: src/app/screens/Home/index.tsx (hunk 3 above)
Comment (lisabaut) [reply src/app/screens/Home/index.tsx:99]: Thanks for the catch. LN data handling is integrated on the AllowanceView too now.
Please check again

**Review by reneaaron — COMMENTED** [review]
Reviewed the code and tested the extension.

@bumi I can't really imagine the effects of our scripts using WebLN inpage (like with the Youtube buttons). But let's take care of that as we get there.

File: src/app/components/TransactionsTable/index.tsx [hunk 5]
```diff
@@ -52,7 +52,19 @@ export default function TransactionsTable({ transactions }: Props) {
                         className="
                       text-sm font-medium text-gray-900 truncate dark:text-white"
                       >
-                        {tx.title || "\u00A0"}
+                        <p className="truncate">
+                          {tx.publisherLink && tx.title ? (
+                            <a
+                              target="_blank"
+                              href={tx.publisherLink}
+                              rel="noreferrer"
```
[the remark this answers is not in the document]
Comment (bumi) [reply src/app/components/TransactionsTable/index.tsx:60]: +1

**Comment by bumi** [comment]
> @bumi I can't really imagine the effects of our scripts using WebLN inpage (like with the Youtube buttons). But let's take care of that as we get there

Yes, we'll see. then it should be pure inpage webln and the page will have an allowance.

File: src/app/components/TransactionsTable/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (lisabaut) [reply src/app/components/TransactionsTable/index.tsx]: done

**Comment by lisabaut** [comment]
@bumi Please approve this PR, otherwise I can not merge because of the "requested changes"

**Comment by escapedcat** [comment]
Looks like we're good here. I'll merge once this is 💚

**Pull request merged**